### ISSUE TITLE
Comply to NC 18+

### ISSUE THREAD

**Issue by DecaTec** [JonathanTreffler/sharerenamer#21]
The app needs to be changed due to changes in NC 18 (share API), see https://github.com/nextcloud/server/pull/15719 and https://github.com/nextcloud/socialsharing/pull/37/commits/931efc9b934ed31e80acd4437f0f253f7e7b1be7.

**Comment by elpraga** [comment]
It would be great if it were made compatible with NC18!

**Comment by small1** [comment]
Just tried to enable it anyway on nc18 and it does not work.

**Comment by DecaTec** [comment]
Yes, it cannot work as the whole share UI was redesigned in NC 18.
Unfortunately, I cannot tell you when I can provide an update of the app.

Any PRs are welcome. :smile:

**Comment by elpraga** [comment]
That is sad to hear. Unfortunately, I cannot contribute with any PRs.

Sincerely, I do not understand why this app isn't part of the official release.

**Comment by elpraga** [comment]
I've just noticed that my previous comment came across a bit harsh.. That was not my intention. I cannot contribute, because I do not know how to code, I'm an English teacher.

Thank you for your work! I hope you find time to patch the app soon :)

**Comment by DecaTec** [comment]
No offense taken. 😃 

Will try to supply a patch soon.

**Comment by warsix4** [comment]
Incredibly useful app extension. I also agree that NC should include this as a default! I am considering reverting my entire NC setup back to 17 purely for this function...

**Comment by adx442** [comment]
> Incredibly useful app extension. I also agree that NC should include this as a default! I am considering reverting my entire NC setup back to 17 purely for this function...

I'm really grateful that previously renamed shares still work in 18 (like my one-way upload folder and some installers that I always send to clients).  DecaTec, I can't help with a PR, but thanks for continuing to work on it.  I miss this functionality in NC every day since 18!

**Comment by Ecphrasis** [comment]
Yes, I really miss this feature too ! Should be official for sure !
Thanks a lot for this, and I hope to use it again soon !

**Comment by morevnaproject** [comment]
Any chance to get this solved soon? ^__^

**Comment by DecaTec** [comment]
I'm working on it, but I'm struggling a but with vue.

**Comment by elpraga** [comment]
Thank you @DecaTec !

**Comment by amenolo** [comment]
This app is gold, thank you @DecaTec for your commitment!

**Comment by amenolo** [comment]
This app is gold, thank you @DecaTec for your effort!

**Comment by stalkerGH** [comment]
Yes, I confirm superb words from other users and waiting for new version of plugin for NC 18 too.

**Comment by linuxmangr** [comment]
YES Is Very useful plugin my users need it! And I waiting it new version.

**Comment by poltrian** [comment]
it would be great if it worked for 18 too. Thanks for your great work

**Comment by stephanedupont** [comment]
Thank you for your work, this is an absolutely vital extension. Fortunately, I saw this before updating Nextcloud, so now I'm waiting for a compatible version before updating from 17.x to 18.x.

**Comment by FelixBuehler** [comment]
Would be very grateful too. Want to update my NC to v18, but I have many shares with customized URL.

**Comment by DecaTec** [comment]
Update:
It will take longer as expected, because on NC 18 there are only limited possibilities to extend the details/share menu. I think that I'll need an extension of NC's API first. I'm already in contact with the NC devs.

**Comment by elpraga** [comment]
Is there a way to support the idea of including the app into Nextcloud?
And, more importantly, would you be interested in doing so @DecaTec ?

**Comment by DecaTec** [comment]
This was already suggested several times.
But the use of the app can lead to a security risk (links might be guessable). Due to this fact, this app will never become "official".

**Comment by szaimen** [comment]
@DecaTec I mean imho it could get included into the core if you would be able to force password protection if a URL is changed with the sharerenamer... (maybe this should also be the default setting?)

**Comment by elpraga** [comment]
I don't think that would be desirable @szaimen There are use case in which using a password is not needed or desired. I believe it should remain up to the admin to decide.

**Comment by adx442** [comment]
> I don't think that would be desirable @szaimen There are use case in which using a password is not needed or desired. I believe it should remain up to the admin to decide.

I think what he's implying is that by adding the _option_ for password protection, it might get it past the Nextcloud gatekeepers.  However, that functionality was already in the past versions.  I agree, though, I certainly don't want mandatory password protection with renamed links.  I've always used it for things like sending someone service files, like https://nc.example.com/s/Teamviewer so it's clear what it is and easy for me to remember and just write in an e-mail without going back to the server to copy the link.

**Comment by cptMikky** [comment]
> But the use of the app can lead to a security risk (links might be guessable).

Wait, seriously? You're saying someone actually considers random links a "security feature"?

**Comment by szaimen** [comment]
> > I don't think that would be desirable @szaimen There are use case in which using a password is not needed or desired. I believe it should remain up to the admin to decide.
> 
> I think what he's implying is that by adding the _option_ for password protection, it might get it past the Nextcloud gatekeepers. However, that functionality was already in the past versions. I agree, though, I certainly don't want mandatory password protection with renamed links. I've always used it for things like sending someone service files, like https://nc.example.com/s/Teamviewer so it's clear what it is and easy for me to remember and just write in an e-mail without going back to the server to copy the link.

No, I just wanted to suggest option to add a configurable share-setting to the admin settings, where the admin can decide what happens with shares that got renamed. I imagine 3 options:
1. Enforce password protection
2. Always ask for a password
3. Disable password protection

I think that this feature would allow the nextcloud team to make this app official.

**Comment by adx442** [comment]
What I was mentioning before is that Password protection already worked with renamed links in the past, it doesn't need to be duplicated in the renamer.

**Comment by Greek64** [comment]
I have to agree with @cptMikky.
Although it is "harder" to guess a random link, this is in no way a security feature, and should in no way be seen as such.
For all practical purposes a random generated link should be treated the same as a custom named link.
If you need protection, that's what the password protection is for (and the password policy enforcer).

If one opens a issue to re-discuss the integration of this app to nextcloud, give me a heads up to cast my vote.

**Comment by szaimen** [comment]
> What I was mentioning before is that Password protection already worked with renamed links in the past, it doesn't need to be duplicated in the renamer.

But I really think that there should be a different setting to "normal" links.
E.g. what if you rename your share to just 
https://nc.example.com/s/a 
Then it is definitely much less secure than a random share.

**Comment by Greek64** [comment]
> E.g. what if you rename your share to just
> https://nc.example.com/s/a
> Then it is definitely much less secure than a random share.

Yeah, and it should be seen exactly as insecure as https://nc.example.com/s/eFmaPRpTsoYJLD3
If you want to share something that needs to be secure, you add a password protection.
Simple as that.

**Comment by szaimen** [comment]
> Yeah, and it should be seen exactly as insecure as https://nc.example.com/s/eFmaPRpTsoYJLD3
> If you want to share something that needs to be secure, you add a password protection.
> Simple as that.
> Edit: Security by obscurity is no security.

What is @DecaTec's opinion on that?

**Comment by DecaTec** [comment]
Well, this is getting a bit off topic right now.
First of all I have to focus on making this app compatible to NC 18.

Please open another issue about that app "becoming official" after it's available on NC 18.

**Comment by Mululu** [comment]
We also use the app very often. Thank you for your efforts.

**Comment by elpraga** [comment]
Any update @DecaTec ?

**Comment by civita** [comment]
I found that I am able to rename the link via Gallery app, and it works and shown back in file app. Not sure this information helps or not. By the way, I added Gallery app in NC 18 by following [this thread](https://help.nextcloud.com/t/new-photo-app-in-nextcloud-18/69949/20).

![Screen Shot 2020-03-17 at 4 59 48 PM](https://user-images.githubusercontent.com/14911217/76839677-e53fa200-6870-11ea-81a9-71550061f0bd.png)

**Comment by DecaTec** [comment]
Sorry, no update so far.

@civita OK, that may work, but it's cumbersome and only a workaround. That's not the way it is intended to be,

**Comment by civita** [comment]
> Sorry, no update so far.
> 
> @civita OK, that may work, but it's cumbersome and only a workaround. That's not the way it is intended to be,

Yes I know, just shared what I found accidently. I used sharerenamer very often (both rename and duplicate links), and thank you for the maintenance. Hope it will work properly in NC 18 soon!

**Comment by cptMikky** [comment]
The question isn't _when will this be done?_

The question is _what can we do to help to make this happen?_

**Comment by castillo92** [comment]
I appreciate so much this app, I use everyday and with NC v18 it does not work. So I cheer developers to make the change as soon as they could.

**Comment by ifuchs** [comment]
Very useful app. I hope that it is made compatible with NC18 soon.

**Comment by e-alfred** [comment]
@DecaTec Maybe @kesselb can help you a bit getting this app back to work?

**Comment by GlenHuf** [comment]
There are a ton of official apps not to include this one by default.   Security risk is non-existant.  Maybe the link could be guessed, who cares,  Maybe add an easy way to list all shares in one place in admin settings.   Problem solved.  Nextcloud's main purpose is to share files, it's just common sense to be able to customize the share.  Some official apps have more security risk than sharerenamer.   Depends how you use the tools you have.

**Comment by w0k3** [comment]
@GlenHuf 

> Well, this is getting a bit off topic right now.
> First of all I have to focus on making this app compatible to NC 18.
> 
> Please open another issue about that app "becoming official" after it's available on NC 18.

**Comment by julpec** [comment]
Hello
I tried to make a (ugly) patch to owncloud 5 or 6 years ago because this feature is really essential especially when working in education: giving a link by writing on the board is so simple and rational!
I have installed a Nextcloud instance in my high school but I always hoped to have this feature so that my colleagues could take advantage of it. I hope that this apps will quickly become an official apps.
Thanks for this work, it's great!

**Comment by evreninanc** [comment]
Hello, first thank you for this great app. Do you have an ETA on latest release compability? Even a rough estimate would do.

Thanks again.

**Comment by e-alfred** [comment]
@DecaTec Maybe you should accept something like Bountysource for your repository so some developer has an incentive to start work on this issue.

**Comment by DecaTec** [comment]
For this app to work again, we need an extension of the NC API.
I'll try to contact the core developers.

**Comment by elpraga** [comment]
Please, do! That would be great!

**Comment by amo13** [comment]
In the meantime, an option to some of us might be a light and open-source link shortener server like [Polr](https://polrproject.org/)

**Comment by DecaTec** [comment]
> For this app to work again, we need an extension of the NC API.
> I'll try to contact the core developers.

They are focused on the upcoming release of NC 19, so this will still last a while.
Sorry...

**Comment by angerized** [comment]
> 
> 
> > For this app to work again, we need an extension of the NC API.
> > I'll try to contact the core developers.
> 
> They are focused on the upcoming release of NC 19, so this will still last a while.
> Sorry...

And after this they will be focused on NC 20, and so on.

Unfortunately NC17 will eventually go EOL, and we will be left with no choice but to either keep outdated software, or upgrade and lose this awesome add-on that you have made.
 
I think we should make our voices heard, how about starting a petition to request support from the Core team? I am sure this would be signed by a lot of people!

**Comment by julpec** [comment]
@angerized +1 !

**Comment by stephanedupont** [comment]
Don't know if it would be useful, but I would sign it for sure if there's a chance it makes it happen.

But doesn't opening an issue on the NC repository is a better way? Is there one already?

**Comment by kesselb** [comment]
@julpec Please use GitHub reactions like everyone else the next time. Everyone subscribed got an notification for your +1 post.

**Comment by kesselb** [comment]
If there are api changes required to make that feature work again open a issue at nextcloud/server and describe that changes. I think bounties are enabled at nextcloud/server.

**Comment by castillo92** [comment]
> > > For this app to work again, we need an extension of the NC API.
> > > I'll try to contact the core developers.
> > 
> > 
> > They are focused on the upcoming release of NC 19, so this will still last a while.
> > Sorry...
> 
> And after this they will be focused on NC 20, and so on.
> 
> Unfortunately NC17 will eventually go EOL, and we will be left with no choice but to either keep outdated software, or upgrade and lose this awesome add-on that you have made.
> 
> I think we should make our voices heard, how about starting a petition to request support from the Core team? I am sure this would be signed by a lot of people!

Are you going to create the petition?  Or we should wait a bit after v19 release?

Keep us update, please! Its a great idea

**Comment by mrizvic** [comment]
Here is ugly low level solution for all you admins out there.

I "renamed" share token for id=2:
```
MariaDB [nc19rc2]> SELECT * FROM oc_share WHERE id=2\G
*************************** 1. row ***************************
              id: 2
      share_type: 3
      share_with: NULL
        password: NULL
       uid_owner: nxcladmin
   uid_initiator: nxcladmin
          parent: NULL
       item_type: folder
     item_source: 989
     item_target: NULL
     file_source: 989
     file_target: /Shared/public
     permissions: 1
           stime: 1589889769
        accepted: 0
      expiration: 2020-06-01 00:00:00
           token: 6NnTBDCdZnj5wRG
       mail_send: 0
      share_name: NULL
password_by_talk: 0
            note: ohai :)
   hide_download: 0
           label:
1 row in set (0.000 sec)
```

Update token field and voila :)
```
MariaDB [nc19rc2]> UPDATE oc_share SET token='myPublicShare' WHERE id=2;
Query OK, 1 row affected (0.001 sec)
Rows matched: 1  Changed: 1  Warnings: 0

MariaDB [nc19rc2]> SELECT * FROM oc_share WHERE id=2\G
*************************** 1. row ***************************
              id: 2
      share_type: 3
      share_with: NULL
        password: NULL
       uid_owner: nxcladmin
   uid_initiator: nxcladmin
          parent: NULL
       item_type: folder
     item_source: 989
     item_target: NULL
     file_source: 989
     file_target: /Shared/public
     permissions: 1
           stime: 1589889769
        accepted: 0
      expiration: 2020-06-01 00:00:00
           token: myPublicShare
       mail_send: 0
      share_name: NULL
password_by_talk: 0
            note: ohai :)
   hide_download: 0
           label:
1 row in set (0.000 sec)
```

**Comment by simaoafonso-pwt** [comment]
Does the `oc_share` table have a constraint for unique `token`s?

**Comment by florianschroen** [comment]
no. just id is a primary key.

```
> show create table oc_share \G
*************************** 1. row ***************************
       Table: oc_share
Create Table: CREATE TABLE `oc_share` (
  `id` bigint(20) NOT NULL AUTO_INCREMENT,
  `share_type` smallint(6) NOT NULL DEFAULT 0,
  `share_with` varchar(255) COLLATE utf8mb4_bin DEFAULT NULL,
  `uid_owner` varchar(64) COLLATE utf8mb4_bin NOT NULL DEFAULT '',
  `uid_initiator` varchar(64) COLLATE utf8mb4_bin DEFAULT NULL,
  `parent` bigint(20) DEFAULT NULL,
  `item_type` varchar(64) COLLATE utf8mb4_bin NOT NULL DEFAULT '',
  `item_source` varchar(255) COLLATE utf8mb4_bin DEFAULT NULL,
  `item_target` varchar(255) COLLATE utf8mb4_bin DEFAULT NULL,
  `file_source` bigint(20) DEFAULT NULL,
  `file_target` varchar(512) COLLATE utf8mb4_bin DEFAULT NULL,
  `permissions` smallint(6) NOT NULL DEFAULT 0,
  `stime` bigint(20) NOT NULL DEFAULT 0,
  `accepted` smallint(6) NOT NULL DEFAULT 0,
  `expiration` datetime DEFAULT NULL,
  `token` varchar(32) COLLATE utf8mb4_bin DEFAULT NULL,
  `mail_send` smallint(6) NOT NULL DEFAULT 0,
  `password` varchar(255) COLLATE utf8mb4_bin DEFAULT NULL,
  `share_name` varchar(64) COLLATE utf8mb4_bin DEFAULT NULL,
  `password_by_talk` tinyint(1) NOT NULL DEFAULT 0,
  `note` longtext COLLATE utf8mb4_bin DEFAULT NULL,
  `hide_download` smallint(6) NOT NULL DEFAULT 0,
  `label` varchar(255) COLLATE utf8mb4_bin DEFAULT NULL,
  PRIMARY KEY (`id`),
  KEY `item_share_type_index` (`item_type`,`share_type`),
  KEY `file_source_index` (`file_source`),
  KEY `token_index` (`token`),
  KEY `share_with_index` (`share_with`),
  KEY `parent_index` (`parent`),
  KEY `owner_index` (`uid_owner`),
  KEY `initiator_index` (`uid_initiator`)
) ENGINE=InnoDB AUTO_INCREMENT=1 DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin ROW_FORMAT=COMPRESSED
```

**Comment by simaoafonso-pwt** [comment]
So if you have duplicated `token`s, what happens? :bomb:

**Comment by mrizvic** [comment]
Share with lower id gets served. But did sharerenamer check for duplicate tokens before? If not then `oc_share`.`token` must be set to UNIQUE to avoid stupidity. But then again how does nextcloud share link generator avoid duplicates?

**Comment by mrizvic** [comment]
It checks for duplicates
https://github.com/DecaTec/sharerenamer/blob/master/db/sharerenamermapper.php#L14

**Comment by Mannshoch** [comment]
If it would help. I should had no problem If you may need to add a named Link to the already existing one eg.: create `nextcloud/s/{ID}` and than add a second link to something like 
`nextcloud/NAMED/{MyNamedLink}`

**Comment by mrizvic** [comment]
Great idea :)

**Comment by Mannshoch** [comment]
If you like to use share renamer to repair a lost linkdestination, this Idea would not help. I remembered I used this on two Links.  May other use it too this way.

**Comment by 8schpi** [comment]
Thank you for the work on sharerenamer @DecaTec. I am waiting with upgrading to Nextcloud18 until there is an update for sharerenamer because it is indispensable for me. The security reasons about sharerenamer (random links = security feature) are not comprehensible, because the data (sharenames) are also random for strangers. If anyone would like more security it is not prohibited to add a secure password.

**Comment by ifuchs** [comment]
Will this ever be made to work with v18?

**Comment by lars-becker** [comment]
It won't help if everybody writes the same. Github has a reaction feature. Please use it and give +1 to views that you share instead of writing the same with different words. Thanks.

**Comment by daschmidt1994** [comment]
some news?

**Comment by DecaTec** [comment]
Unfortunately not. If NC's API isn't expanded, I guess that I have to drop support for this app.
I'm really sorry to say that.

**Comment by stephanedupont** [comment]
Is there an issue somewhere (or a feature request) on Nextcloud repository asking for this, so we can all vote for it and weigh in?

**Comment by Mannshoch** [comment]
What's with the Idea https://github.com/DecaTec/sharerenamer/issues/21#issuecomment-638383723 ?

**Comment by DecaTec** [comment]
Sadly, I cannot maintain this project any further.
I don't have the time to make the app compatible to Nextcloud 18+ as it probably will need some changes in Nextcloud's core code.

I'm sorry that you all waited so long for an update which will probably never come.

This way, I'm looking for a new maintainer, see #29.

**Comment by nursoda** [comment]
Thanks for your decision, it's clear now what (not) to expect.
I can fully understand that life goes on and focus change.
THANK YOU for what you did so far and all the best for you.

@Nextcloud devs: This App should be a core function, not an app, really.

**Comment by Mannshoch** [comment]
May someon add this to:
https://help.nextcloud.com/t/nextcloud-deck-social-and-forms-were-listening-to-your-thoughts-and-feature-wishes/82914/1

**Comment by JonathanTreffler** [comment]
It looks like i am going to continue the development of this App :) #29 

I will be able to start developing in a few weeks (maybe sooner).

**Comment by elpraga** [comment]
That's great news!!

**Comment by djetaine** [comment]
> Here is ugly low level solution for all you admins out there.
> 
> I "renamed" share token for id=2:
> 
> ```
> MariaDB [nc19rc2]> SELECT * FROM oc_share WHERE id=2\G
> *************************** 1. row ***************************
>               id: 2
>       share_type: 3
>       share_with: NULL
>         password: NULL
>        uid_owner: nxcladmin
>    uid_initiator: nxcladmin
>           parent: NULL
>        item_type: folder
>      item_source: 989
>      item_target: NULL
>      file_source: 989
>      file_target: /Shared/public
>      permissions: 1
>            stime: 1589889769
>         accepted: 0
>       expiration: 2020-06-01 00:00:00
>            token: 6NnTBDCdZnj5wRG
>        mail_send: 0
>       share_name: NULL
> password_by_talk: 0
>             note: ohai :)
>    hide_download: 0
>            label:
> 1 row in set (0.000 sec)
> ```
> 
> Update token field and voila :)
> 
> ```
> MariaDB [nc19rc2]> UPDATE oc_share SET token='myPublicShare' WHERE id=2;
> Query OK, 1 row affected (0.001 sec)
> Rows matched: 1  Changed: 1  Warnings: 0
> 
> MariaDB [nc19rc2]> SELECT * FROM oc_share WHERE id=2\G
> *************************** 1. row ***************************
>               id: 2
>       share_type: 3
>       share_with: NULL
>         password: NULL
>        uid_owner: nxcladmin
>    uid_initiator: nxcladmin
>           parent: NULL
>        item_type: folder
>      item_source: 989
>      item_target: NULL
>      file_source: 989
>      file_target: /Shared/public
>      permissions: 1
>            stime: 1589889769
>         accepted: 0
>       expiration: 2020-06-01 00:00:00
>            token: myPublicShare
>        mail_send: 0
>       share_name: NULL
> password_by_talk: 0
>             note: ohai :)
>    hide_download: 0
>            label:
> 1 row in set (0.000 sec)
> ```

Thanks for this. I'm using postgres but the process is basically the same. Things are much prettier now! Sure, I have to do it manually for share links, but it only takes a second. Very good workaround until the UI gets added.

**Comment by skacco** [comment]
hello any news?

**Comment by JonathanTreffler** [comment]
Not yet. I am currently in grade 12 (wich is the last one in germany) and have to focus on school for a bit. But its quite high on my priority list :)

**Comment by Tuffiglig** [comment]
Hey @JonathanTreffler 
Sorry to ask the same question again.
Do you have a rough guess when you have time to look into this project?
With release of Nextcloud 17.0.10
[](https://nextcloud.com/changelog/#latest17)
and the note: this is the final 17 release.
it is not a good idea to stay on version 17.

**Comment by JonathanTreffler** [comment]
No problem.
I will write a german test tomorrow and a astrophysics test next week. After that there are holidays, but i can't promise i have time then (there are 5 other tests immediately after the holidays).

**Comment by JonathanTreffler** [comment]
I started the PR #31 to this repository, where i will implement the rewrite.
I hope i will be able to get a working version of the rewrite within the next 2 weeks.

**Comment by roberthr74** [comment]
Any news on this plugin?

**Comment by Mululu** [comment]
I assume that nothing more will come here !?

**Comment by H1lseduck** [comment]
If you want custom links NOW, just go to your Database (used for Nextcloud), e.g. with phpMyAdmin, and then search for "oc_share". In oc_share, you see all the active sharing links, like their destination, their creator, and their creation date, etc.
Now just look up the link you want to change the ending of, and just change the value in the "token" column to whatever you'd like.
If you changed it to e.g. testtest, you can now access the shared file via https://yourdomain.com/nextcloud/index.php/s/testtest

Of course a bit of work and completely inefficient, but better than nothing.

**Comment by nursoda** [comment]
I'd recommend a Webserver redirect to the (generated) share link. That way, no db fiddling, plus links get even shorter, e.g.
https://yourdomain.com/shorturl → https://cloud.yourdomain.com/s/generatedrandom

**Comment by castillo92** [comment]
> If you want custom links NOW, just go to your Database (used for Nextcloud), e.g. with phpMyAdmin, and then search for "oc_share". In oc_share, you see all the active sharing links, like their destination, their creator, and their creation date, etc.
> Now just look up the link you want to change the ending of, and just change the value in the "token" column to whatever you'd like.
> If you changed it to e.g. testtest, you can now access the shared file via https://yourdomain.com/nextcloud/index.php/s/testtest
> 
> Of course a bit of work and completely inefficient, but better than nothing.

And why this can not be automated like did DecaTec?

**Comment by panfass** [comment]
This plugin is the main reason holding me back to version 17. Hoping someday it will be supported for current NC versions :)

**Comment by szaimen** [comment]
> This plugin is the main reason holding me back to version 17.

You know that NC17 is eol since over half a year?

**Comment by djetaine** [comment]
I made a quick tutorial video on how to do this in sql for someone on reddit a while back if anyone is interested. 
https://www.youtube.com/watch?v=UaMzAuXvmuk

**Comment by gaellafond** [comment]
@DecaTec We need this feature on NC 21. My team and I have time to help with the development.

- Where did you manage to get with the development?
- What is missing in NC API to finish the work?
- Did you get feedback from NC team?

**Comment by DecaTec** [comment]
@gaellafond Unfortunately, I cannot maintain the app any further.
But maybe a new maintainer was found.
@JonathanTreffler Are you still planning to rewrite/maintain the app in the future?

**Comment by gaellafond** [comment]
@DecaTec Thanks for the quick response. I'm about to check-out the source and try to understand it. I'm new to NextCloud but I have many years of experience in programming. I have hope that I can fix it.

I can see there is a "Nextcloud-18" branch and a "v3" branch. Do you remember which of those branch contains the most recent development? I was thinking of looking at "Nextcloud-18" first.

**Comment by DecaTec** [comment]
I think "v3" is the rewrite of @JonathanTreffler 
But I don't know if it was just a try or if there was addes something woth to build on.

As a hint: The app (as it is) fiddles around with the DOM with JS. This cannot be done anymore, because NC changed the UI to vue.js. So probably, you have to write a whole new vue component extending the "details pane" in NC.

**Comment by gaellafond** [comment]
@DecaTec Thanks, I understand the issue now. I have good experience with JS but I have never used Vue. I will wait to see what @JonathanTreffler have to say.

**Comment by JonathanTreffler** [comment]
I plan to continue my pr and my Abitur (equivalent to final exam of high school in germany) is done, but i broke my foot and cannot currently work productively. I have a lot of projects, and while this is not the my top priority it is quite high on my todo list :)

**Comment by JonathanTreffler** [comment]
> As a hint: The app (as it is) fiddles around with the DOM with JS. This cannot be done anymore, because NC changed the UI to vue.js. So probably, you have to write a whole new vue component extending the "details pane" in NC.

Exactly, thats what i plan to do in the v3 rewrite.

**Comment by stephanedupont** [comment]
Well, @JonathanTreffler your last contribution was October 2020 and your last message doesn't soon too optimistic about doing it quickly, so personally I'm all for @DecaTec and his team taking up the torch :)

**Comment by JonathanTreffler** [comment]
Well i would predict, that i can continue the work in about a week.
If somebody wants to help he can obviously do that. The majority of the work is already done in the v3 branch, so getting it done should not take long.

**Comment by szaimen** [comment]
There is now an api which makes it possible to inject entries into the share dropdown:
https://github.com/nextcloud/server/pull/28835

**Comment by gaellafond** [comment]
> It will be available in NC23 and NC22.2.0

I download the `files_downloadlimit` app, compiled it, install it and when I tried to enable it, NextCloud said it was for NC 23. I'm so disappointed. I guess I will have to try the latest version now...

**Comment by szaimen** [comment]
Please report this to the correct repo (it shall work on NC22 in the future, too) Thanks!

**Comment by JonathanTreffler** [comment]
> There is now an api which makes it possible to inject entries into the share dropdown:
> https://github.com/nextcloud/server/pull/28835
> It will be available in NC23 and NC22.2.0

Wow, cool

**Comment by JonathanTreffler** [comment]
First we need to get the NC 20-22 compatible version of this app out there, but after that we could drop support for NC 20-21 and implement it with the new API.

**Comment by Memphizzz** [comment]
Any news on this?

**Comment by gaellafond** [comment]
@Memphizzz JonathanTreffler created a working version of the app. I think it's only waiting approval. A few Lint tests are still failing.

To install the plugin (assuming you do not want to compile it):
Download it from here:
https://github.com/JonathanTreffler/sharerenamer/files/7216628/sharerenamer.zip

The code is available in branch "v3". See pull requests for more info:
https://github.com/JonathanTreffler/sharerenamer/pull/31#issuecomment-925671498-permalink

**Comment by JonathanTreffler** [comment]
> @Memphizzz JonathanTreffler created a working version of the app. I think it's only waiting approval. A few Lint tests are still failing.

The limiting factor is my time right now. The only things left to do is request a new certificate for the nextcloud app store and upload it there. Sadly I can't give an ETA right now, but it is very easy to manually install the v3 version, so nobody has to wait until it is pushed to the appstore.

**Comment by julpec** [comment]
I can't get the application to work. I am under Nextcloud version 22.2.2 and I install with this procedure :
```
cd /var/www/nextcloud
git clone https://github.com/JonathanTreffler/sharerenamer/
cd sharerenamer
git checkout v3
cd ..
chown -R www-data:www-data sharerenamer
```

After that, I activate the application in the administration interface. After that, I have no reference to the renaming of the public link.
Did I make a mistake?

**Comment by JonathanTreffler** [comment]
> I can't get the application to work. I am under Nextcloud version 22.2.2 and I install with this procedure :
> 
> ```
> cd /var/www/nextcloud/apps
> git clone https://github.com/JonathanTreffler/sharerenamer/
> cd sharerenamer
> git checkout v3
> cd .. && chown -R www-data:www-data sharerenamer
> ```
> 
> After that, I activate the application in the administration interface. After that, I have no reference to the renaming of the public link. Did I make a mistake? Or I didn't understand if the app in v3 was now compatible with NC22. The thread is not very clear about this.
> 
> Thanks

You need to compile it first. I would recommend downloading the [zip file](https://github.com/JonathanTreffler/sharerenamer/files/7216628/sharerenamer.zip) mentioned in this issue, which contains a compiled pre release.

**Comment by julpec** [comment]
Very well, thank you.
So, [this section](https://github.com/JonathanTreffler/sharerenamer#install-from-git) must be modified in the readme.
Also, it doesn't work. I extracted the .zip in apps/sharerenamer but still no visible features after activation of the app.

**Comment by JonathanTreffler** [comment]
Have you looked in the share tab or in the sharerenamer tab ? (Previously the sharerenamer app injected itself into the share tab, it now has its own tab)

**Comment by JonathanTreffler** [comment]
@DecaTec  I need your help to transfer the apps permission on the AppStore: https://nextcloudappstore.readthedocs.io/en/latest/developer.html#transferring-your-app-to-a-new-owner

Can you please unlock sharerenamer ?
I have already applied for a new certificate in https://github.com/nextcloud/app-certificate-requests/pull/478.

**Comment by julpec** [comment]
Unfortunately, on a NC server v21, there is no specific tab of the application, nor any entry related to this application that I can detect
BUT, on a NC v22 server I can see this tab and I can rename my share !
However, I noticed a small flaw in the functionality. When you create a public share, you can't rename it right away. You have to reload the page to have access to the modification in the sharerenamer tab.
In conclusion, it works on a v22 server, sharerenamer have his own tab, but it is not possible to rename on the fly.

**Comment by JonathanTreffler** [comment]
Ok, I will then only release it for Nextcloud 22.
With Nextcloud 23 already out I don't have the time to fix it in 21.
Thanks for looking into it.

**Comment by JonathanTreffler** [comment]
I just tested it in Nextcloud 23 and it worked :)
So the release (which I hope to release in the next days) will include support for Nextcloud 22 and 23.

**Comment by JonathanTreffler** [comment]
This got automatically closed, because I merged the pull request, but I am going to open it until the release for the app store is done.

**Comment by jshatch** [comment]
I just tried the pre-release zip and it works.  If you create a new share on a file the new share doesn't show up in the rename list until you close and reopen the sidebar.  Same thing with renaming a share, the new share name does not reflect in the link on the sharing tab until the sidebar is closed and reopened.  Maybe not a critical issue, but something to be aware of.

**Comment by JonathanTreffler** [comment]
> I just tried the pre-release zip and it works. If you create a new share on a file the new share doesn't show up in the rename list until you close and reopen the sidebar. Same thing with renaming a share, the new share name does not reflect in the link on the sharing tab until the sidebar is closed and reopened. Maybe not a critical issue, but something to be aware of.

I am aware of that, it is something I will address in a later release. Right now I just want to get the update out as fast as possible :)

**Comment by JonathanTreffler** [comment]
Good News:
I have gotten a certificate from Nextcloud and the ownership of the app in the app store has been transferred to me, which means there's nothing standing in the way of releasing v3 to the app store 🎉🎉🎉🎉🎉🎉🎉🎉

Tomorrow I will update the README and all the descriptions and release v3 to the app store.

Thanks for your patience and thanks to everyone, who tried the "alpha" release.

**Comment by JonathanTreffler** [comment]
New Screenshots :)

![sharerenamer](https://user-images.githubusercontent.com/28999431/148477750-7e0fce72-c90f-48aa-9d68-2621bf778d49.png)
![sharerenamer2](https://user-images.githubusercontent.com/28999431/148477746-f4968483-7de0-476a-ac4a-913bf255654f.png)

**Comment by DecaTec** [comment]
This is great news!
Thanks a lot for your efforts!
Looking forward for the app being released on the NC app store soon.

**Comment by JonathanTreffler** [comment]
The v3.0.0 release is published on [Github](https://github.com/JonathanTreffler/sharerenamer/releases/tag/v3.0.0)  🎉

I will now upload the release to the Nextcloud app store. I don't know if that will take time to process, but I will post here, once it can be downloaded through the store.

**Comment by JonathanTreffler** [comment]
Version 3.0.0 can now be downloaded through the app store :)

**Comment by JonathanTreffler** [comment]
In my nextcloud instance sharerenamer is now in the app store list :)
I will close this now.
